fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRRestHelper.java

```diff
@@ -0,0 +1,2448 @@
+        Bundle inputBundle = null;
+
+        // Save the current request context.
+        FHIRRequestContext requestContext = FHIRRequestContext.get();
+
+        try {
+            if (bundleResource instanceof Bundle) {
+                inputBundle = (Bundle) bundleResource;
+            } else {
+                String msg = "A 'Bundle' resource type is required but a '"
+                        + bundleResource.getClass().getSimpleName() + "' resource type was sent.";
+                throw buildRestException(msg, IssueType.INVALID);
+            }
+            // First, validate the bundle and create the response bundle.
+            Bundle responseBundle = validateBundle(inputBundle);
+
+            // Next, process each of the entries in the bundle.
+            responseBundle = processBundleEntries(inputBundle, responseBundle, requestProperties);
+
+            return responseBundle;
+        } finally {
+            // Restore the original request context.
+            FHIRRequestContext.set(requestContext);
+
+            log.exiting(this.getClass().getName(), "doBundle");
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * @see com.ibm.fhir.rest.FHIRResourceHelpers#getTransaction()
+     */
+    @Override
+    public FHIRPersistenceTransaction getTransaction() throws Exception {
+        return persistence.getTransaction();
+    }
+
+    /**
+     * Validate the input resource and throw if there are validation errors
+     *
+     * @param resource
+     * @throws FHIRValidationException
+     * @throws FHIROperationException
+     */
+    private List<OperationOutcome.Issue> validateInput(Resource resource)
+            throws FHIRValidationException, FHIROperationException {
+        List<OperationOutcome.Issue> issues = FHIRValidator.validator().validate(resource);
+        if (!issues.isEmpty()) {
+            boolean includesFailure = false;
+            for (OperationOutcome.Issue issue : issues) {
+                if (FHIRUtil.isFailure(issue.getSeverity())) {
+                    includesFailure = true;
+                }
+            }
+
+            if (includesFailure) {
+                throw new FHIROperationException("Input resource failed validation.").withIssue(issues);
+            } else {
+                if (log.isLoggable(Level.FINE)) {
+                    String info = issues.stream()
+                                .flatMap(issue -> Stream.of(issue.getDetails()))
+                                .flatMap(details -> Stream.of(details.getText()))
+                                .flatMap(text -> Stream.of(text.getValue()))
+                                .collect(Collectors.joining(", "));
+                    log.fine("Validation warnings for input resource: " + info);
+                }
+            }
+        }
+        return issues;
+    }
+
+    /**
+     * @param issues
+     * @return
+     */
+    private boolean anyFailureInIssues(List<OperationOutcome.Issue> issues) {
```

Comment:
This is going to sound strange... I think this one has validity to move up to FHIRUtil
I think I wrote similar code for tests (not only for tests) in Search

Reply:
I actually agree.  Previously it was in FHIRResource so I just moved it over along with the doX methods since thats where it was used.

Reply:
Agree on moving to FHIRUtil

Reply:
or the fhir-validator project

Reply:
I didn't do this...but I can.  Should we do it now or accept this PR without it?